pkcs11/src/new/mod.rs

```diff
@@ -2,13 +2,14 @@ pub mod functions;
 pub mod objects;
 pub mod types;
 
-use pkcs11_sys::CKR_OK;
+use crate::new::types::function::Rv;
 use std::mem;
 use std::path::Path;
 
 pub struct Pkcs11 {
-    //this field is not needed once we have the function list
-    //pkcs11_lib: pkcs11_sys::Pkcs11,
```

Comment:
We learn everyday 🤣

Reply:
😱  what? How?

Reply:
I am not exactly sure. I tried to remove the field and then I had segfaults. I think the libloading structure contained in `pkcs11_sys::Pkcs11` actually contains the memory where the dynamic functions are loaded? So if we don't store it somewhere, the memory is released and the function pointers point at invalid memory.

Is that something like that @joechrisellis?

Reply:
Hello!

I am not an expert on how libloading works but that sounds about right to me. `pkcs11_sys::Pkcs11` owns the `libloading::Library` structure, which presumably owns some chunk of memory containing the loaded library functions. If you don't own the `pkcs11_sys::Pkcs11` struct inside the `Pkcs11` struct here, the `Pkcs11_sys::Pkcs11` (and by extension its `libloading::Library`) will be dropped. So it would be undefined behaviour to try accessing this memory. :)

Reply:
Thanks! It's a private field, there shouldn't be any issue in just keeping it there.